$NetBSD: patch-ad,v 1.6 2011/12/20 18:33:40 ryoon Exp $

Two unrelated things configure needs.

Machines and systems it has never heard of: alpha, powerpc and x86_64 on
NetBSD, and DragonFly.

Its test for whether the C preprocessor works checks that nothing came out
on stderr.  A preprocessor that says anything at all -- a warning, a note
-- fails the test even though it works.  Check that the output holds what
the input should have produced instead.

--- configure.orig	1995-06-05 01:11:38.000000000 +0000
+++ configure
@@ -581,13 +581,16 @@ case "${canonical}" in
   *-*-netbsd* )
     opsys=netbsd
     case "${canonical}" in
+      alpha-*-netbsd*)	machine=alpha ;;
       i[345]86-*-netbsd*) machine=intel386 ;;
       m68k-*-netbsd*)
 			# This is somewhat bogus.
 			machine=hp9000s300 ;;
       mips-*-netbsd*)	machine=pmax ;;
       ns32k-*-netbsd*)	machine=ns32000 ;;
+      powerpc-*-netbsd*) machine=powerpc ;;
       sparc-*-netbsd*)	machine=sparc ;;
+      x86_64-*-netbsd*) machine=amd64 ;;
     esac
   ;;
 
@@ -1177,6 +1180,7 @@ case "${canonical}" in
       *-bsd386* | *-bsdi* )	opsys=bsd386 ;;
       *-386bsd* )	        opsys=386bsd ;;
       *-freebsd* )	        opsys=freebsd ;;
+      *-dragonfly* )		opsys=dragonfly ;;
       *-nextstep* )             opsys=mach2 ;;
       ## Otherwise, we'll fall through to the generic opsys code at the bottom.
     esac
@@ -1384,10 +1388,13 @@ if test -z "$CPP"; then
 #include "confdefs.h"
 #include <stdio.h>
 Syntax Error
+Should \\
+be one line.
 EOF
 # Some shells (Coherent) do redirections in the wrong order, so need
 # the parens.
-ac_err=`eval "($ac_cpp conftest.${ac_ext} >/dev/null) 2>&1"`
+ac_err=`eval "($ac_cpp conftest.${ac_ext} > conftest2.out) 2>&1 ; \
+grep -L \"Should  *be one line\" conftest2.out"`
 if test -z "$ac_err"; then
   :
 else
@@ -1397,10 +1404,13 @@ else
 #include "confdefs.h"
 #include <stdio.h>
 Syntax Error
+Should \\
+be one line.
 EOF
 # Some shells (Coherent) do redirections in the wrong order, so need
 # the parens.
-ac_err=`eval "($ac_cpp conftest.${ac_ext} >/dev/null) 2>&1"`
+ac_err=`eval "($ac_cpp conftest.${ac_ext} >/dev/null) 2>&1 ; \
+grep -L \"Should  *be one line\" conftest2.out"`
 if test -z "$ac_err"; then
   :
 else
